clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -1343,6 +1343,9 @@ namespace {
     DeclarationName Entity;
     // Whether to evaluate the C++20 constraints or simply substitute into them.
     bool EvaluateConstraints = true;
+    // Whether Substitution was Incomplete, that is, we tried to substitute in
+    // any template arguments which were null.
```

Comment:
Beecause they were not yet provided, presumably?

Reply:
Yes, if the user passes any null template arguments, and we try to use any of these arguments during substitution, then this flag will be set, which will help determine that the result of this substitution is invalid and should not be looked at.